src/app/code/community/Omise/Gateway/Model/Payment/Offsitepaynow.php

```diff
@@ -0,0 +1,162 @@
+<?php
+class Omise_Gateway_Model_Payment_Offsitepaynow extends Omise_Gateway_Model_Payment_SimpleOffsite_Payment
+{
+
+    /**
+     * @var string
+     */
+    protected $_code = 'omise_offsite_paynow';
+
+    /**
+     * @var string
+     */
+    protected $_formBlockType = 'omise_gateway/form_default';
+
+    /**
+     * @var string
+     */
+    protected $_infoBlockType = 'payment/info';
+
+    /**
+     * @var string
+     */
+    protected $_callbackUrl = 'omise/callback_validateoffsitepaynow';
+
+    /**
+     * @var array
+     */
+    protected $_currencies = array('SGD');
+
+    /**
+     * @var string
+     */
+    protected $_type = 'paynow';
+
+    /**
+     * Payment Method features
+     *
+     * @var bool
+     */
+    protected $_isGateway          = true;
+    protected $_canReviewPayment   = true;
+    protected $_isInitializeNeeded = true;
+    protected $_allowRedirect      = false;
+
+    /**
+     * Instantiate state and set it to state object
+     *
+     * @param string        $payment_action
+     * @param Varien_Object $state_object
+     */
+    public function initialize($payment_action, $state_object) {
+        $payment = $this->getInfoInstance();
+        $order   = $payment->getOrder();
+        $invoice = $order->prepareInvoice()->register();
+        $charge = $this->processPayment($payment, $invoice->getBaseGrandTotal());
+        $payment->setCreatedInvoice($invoice)
+            ->setIsTransactionClosed(false)
+            ->setIsTransactionPending(true)
+            ->addTransaction(
+                Mage_Sales_Model_Order_Payment_Transaction::TYPE_CAPTURE,
+                $invoice,
+                false,
+                Mage::helper('omise_gateway')->__('Capturing an amount of %s via '.$this->getPaymentTitle().'.', $order->getBaseCurrency()->formatTxt($invoice->getBaseGrandTotal()))
+            );
+        $order->addRelatedObject($invoice);
+        if ($charge->isAwaitPayment() || $charge->isAwaitCapture() || $charge->isSuccessful()) {
+            $state_object->setState(Mage_Sales_Model_Order::STATE_PENDING_PAYMENT);
+            $state_object->setStatus($order->getConfig()->getStateDefaultStatus(Mage_Sales_Model_Order::STATE_PENDING_PAYMENT));
+            if($charge->isAwaitPayment() ) {
+                $state_object->setIsNotified(true);
+                $this->sendOrderConfirmationEmail($order, $charge);
+                $payment->setIsTransactionPending(true);
+                Mage::getSingleton('checkout/session')->setOmiseAuthorizeUri(Mage::getUrl('omise/checkout_paynow', array()));
+            }
+            
+            return;
+        }
+        $this->_suspectToBeFailed($payment);
+    }
+
+    /**
+     * @param Varien_Object $payment
+     * @param float $amount 
+     *
+     * @return Omise_Gateway_Model_Api_Charge
+     * @throws Mage_Core_Exception
+     */
+    public function processPayment(Varien_Object $payment, $amount) {
```

Comment:
This function doesn't need to be here - it is identical in the parent class. Weirdly though - it has a different name in the parent class. The same applies to Tesco and TrueMoney. Something feels quite wrong here